### ISSUE TITLE
SparseTensor Error! Indices are not valid (out of bounds)

### ISSUE THREAD

**Issue by unlabeledData** [Stick-To/CenterNet-tensorflow#22]
Hi, I try your code under tensorflow 1.4 and train on COCO dataset. I met some bugs and have no idea to solve it. Could you give me a hand?

In TF 1.4, `tf.boolean_mask doesn't support axis, so I remove it.
For the same TF version reason, I use tensor.sparse_tensor_to_dense and tf.SparseTensor.

>InvalidArgumentError (see above for traceback): Indices are not valid (out of bounds).  Shape: [128,128]
	 [[Node: get_loss/cond_649/SparseToDense = SparseToDense[T=DT_FLOAT, Tindices=DT_INT64, validate_indices=false, _device="/job:localhost/replica:0/task:0/device:CPU:0"](get_loss/cond_649/strided_slice/Switch, get_loss/cond_649/SparseToDense/Switch/_15033, get_loss/cond_649/ones_like/_15035, get_loss/cond_649/SparseToDense/default_value)]]

The codes are:
 `        for i in range(self.num_classes):

            i = tf.Print(i, [i], 'i')
            exist_i = tf.equal(classid, i)
            reduce_i = tf.boolean_mask(
                keyp_penalty_reduce, exist_i)
            reduce_i = tf.Print(
                reduce_i, [tf.shape(reduce_i)], 'tf.shape(reduce_i)')
            reduce_i = tf.cond(
                tf.equal(tf.shape(reduce_i)[0], 0),
                lambda: zero_like_keyp,
                lambda: tf.expand_dims(
                    tf.reduce_max(reduce_i, axis=0), axis=-1)
            )
            reduction.append(reduce_i)

            gbbox_yx_i = tf.boolean_mask(gbbox_yx, exist_i)
            gbbox_yx_i = tf.Print(
                gbbox_yx_i, [tf.shape(gbbox_yx_i)], 'tf.shape(gbbox_yx_i)')
            pshape = tf.to_int64(pshape)

            gt_keypoints_i = tf.cond(
                tf.equal(tf.shape(gbbox_yx_i)[0], 0),
                lambda: zero_like_keyp,
                lambda: tf.expand_dims(tf.sparse_tensor_to_dense(tf.SparseTensor(gbbox_yx_i, tf.ones_like(
                    gbbox_yx_i[..., 0], tf.float32), dense_shape=pshape), validate_indices=False), axis=-1)
                # sparse_to_dense version
            )
            gt_keypoints.append(gt_keypoints_i)`

**Comment by unlabeledData** [comment]
I solved this problem. The parameter `pad_truth_to` in image_augmentor.py can not be `None`.
